persistence-dse-6.8/src/main/java/io/stargate/db/dse/impl/DsePersistence.java

```diff
@@ -352,7 +353,10 @@ protected void loginInternally(io.stargate.db.AuthenticatedUser user) {
         // at least some user to be present but you cannot login with the "system" user.
         return Single.just(
             new QueryState(
-                ClientState.forExternalCalls(AuthenticatedUser.ANONYMOUS_USER),
+                new ClientStateWithPublicAddress(
+                    AuthenticatedUser.ANONYMOUS_USER,
+                    new InetSocketAddress("127.0.0.1", 0),
+                    new InetSocketAddress("127.0.0.1", 0)),
                 UserRolesAndPermissions.SYSTEM));
       } else {
```

Comment:
Should we use `ClientStateWithPublicAddress` in the `else` branch too?

Reply:
Good catch, was assuming it worked but looks like it doesn't.

Reply:
Given that `ProxyProtocolQueryInterceptor` expects all client states to be `ClientStateWithPublicAddress` I wonder if we should convert all places when `ClientState` is created :thinking: